spatialprofilingtoolbox/workflow/common/sparse_matrix_puller.py

```diff
@@ -161,61 +170,57 @@ def fill_data_arrays_for_study(
             progress_reporter.increment(iteration_details=_specimen)
         progress_reporter.done()
 
-    def get_pertinent_specimens(self, study_name, specimen: str=None):
+    def _get_pertinent_specimens(self, study_name: str, specimen: str | None=None):
         if specimen is not None:
             return [specimen]
-        with self.get_connection().cursor() as cursor:
-            cursor.execute('''
-            SELECT sdmp.specimen
-            FROM specimen_data_measurement_process sdmp
-            WHERE sdmp.study=%s
-            ORDER BY sdmp.specimen
-            ;
-            ''', (study_name,))
-            rows = cursor.fetchall()
+        self.cursor.execute('''
+        SELECT sdmp.specimen
+        FROM specimen_data_measurement_process sdmp
+        WHERE sdmp.study=%s
+        ORDER BY sdmp.specimen
+        ;
+        ''', (study_name,))
+        rows = self.cursor.fetchall()
         return [row[0] for row in rows]
 
-    def get_study_names(self, connection, study=None):
+    def _get_study_names(self, study: str | None=None) -> list[str]:
```

Comment:
Why did this internal function get a return type hint when most pubic functions didn't?

Reply:
Yes they should have them, I added a few more type hints.
The answer is that this module was written without regard to type hinting. Every type hint was added after the module was written.